lib/ejs.js

```diff
@@ -351,6 +360,10 @@ exports.renderFile = function () {
     else {
       cpOptsInData(data, opts);
     }
+    // express auto 'root' option;
+    if (data && data.settings && type(data.settings.views) == "string") {
+      opts.root = data.settings.views;
+    }
```

Comment:
Express-specific code should not be in `renderFile`, but rather in the Express function only.

Also, this code breaks compatibility, as the default include resolving mechanism is changed by this block, so in fact it shouldn't be there at all.

Also, this code is apparently untested, since the `type` function is undefined.